code/modules/admin/verbs/debug.dm

```diff
@@ -871,3 +871,42 @@ GLOBAL_PROTECT(LastAdminCalledProc)
 	message_admins("<span class='adminnotice'>[key_name_admin(src)] pumped a random event.</span>")
 	SSblackbox.add_details("admin_verb","Pump Random Event")
 	log_admin("[key_name(src)] pumped a random event.")
+
+/client/proc/start_line_profiling()
+	set category = "Profile"
+	set name = "Start Line Profiling"
+	set desc = "Starts tracking line by line profiling for code lines that support it"
+
+	PROFILE_START
+
+	message_admins("<span class='adminnotice'>[key_name_admin(src)] started line by line profiling.</span>")
+	SSblackbox.add_details("admin_verb","Start Line Profiling")
+	log_admin("[key_name(src)] started line by line profiling.")
+
+/client/proc/stop_line_profiling()
+	set category = "Profile"
+	set name = "Stops Line Profiling"
+	set desc = "Stops tracking line by line profiling for code lines that support it"
+
+	PROFILE_STOP
+
+	message_admins("<span class='adminnotice'>[key_name_admin(src)] stopped line by line profiling.</span>")
+	SSblackbox.add_details("admin_verb","stop Line Profiling")
+	log_admin("[key_name(src)] stopped line by line profiling.")
+
+/client/proc/show_line_profiling()
```

Comment:
can we apply this to all the A* tests

Reply:
no

Reply:
why not, wouldn't this let us figure out where byond is shitting itself on a*

Reply:
There is enough overhead from adding this that it wouldn't be appropriate to add it to a proc like that in production.

Lesser used systems that need enhancement could benefit

And you can always use this in testing

Reply:
Okay, that makes sense.

Reply:
I know, where it is fucked up.It is fucked by byond magic.Lummox told, that access to datum var is slower, than list fuckery.
Maybe I'll rewrite it in way, which Lummox proposed.But code will become a lot more ugly,yet performant.